README.md

```diff
@@ -163,6 +163,27 @@ retry_options = {
 }
 ```
 
+### Call a block after retries have been exhausted
+
+You can specify a lambda object through the `exhausted_retries_block` option that will be called after all retries are exhausted.
+This block will be called once.
+
+The block is passed keyword arguments with contextual information and passed your data:
+* Request environment,
+* exception,
+* middleware options
+* and your data.
+
+In a lambda you can pass any logic for further work.
+
+For example, you might want to update user by request query.
+
+```ruby
+retry_options = {
+  exhausted_retries_block: -> (user_id:, env:, exception:, options:) { User.find_by!(id: user_id).do_admin! }
```

Comment:
This documentation may be outdated, it says "your data" will be passed too, but looking at the implementation that doesn't seem to be the case.
You can indeed add more keyword arguments if you want to (as you do in the specs below), but those would require a default value so in practice this would work?

Reply:
I thought the tests covered this point.

Reply:
If you're referring to the `numbers: 1` argument in tests, then yeah that works because you're passing a default value, but in this case `user_id` has no default value, so I suspect this example won't work?
Unless I'm missing something and `user_id` gets injected somehow?

Reply:
No, Matt. It's just an example for the documentation.

Do you think it's better to use user_id for consistency?

Reply:
Nah is fine, I just wanted to make sure I understand how the feature works 🙂